Chat with Azure OpenAI quickstart sample (#285)
Azure-Samples/communication-services-javascript-quickstarts: chat-openai-sample/README.md

|page_type|languages|products
|---|---|---|
|sample|<table><tr><td>Typescript</tr></td></table>|<table><tr><td>azure</td><td>azure-communication-services</td></tr></table>|

> # Chath with Azure OpenAI - Quick Start Sample

Reviewer: ```suggestion
# Chat with Azure OpenAI - Quick Start Sample
```

> This sample application demonstrates how to integrate Azure Communication Services Chat SDK with Azure OpenAI Service to enable intelligent message analysis. The application listens for a user message, processes the text through the Azure OpenAI Service, and generates appropriate analysis. Or optionally, developers can replace the logic with their own AI model for message analysis.

Reviewer 2: As part of the intro or subsection, it would be helpful to describe the components. For example:

- app.ts - Node JS application providing HTTP endpoints for message analysis
- client.ts - script to setup chat messages and test HTTP endpoints for message analysis

- An Azure OpenAI Resource and Deployed Model. See [instructions](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/create-resource?pivots=web-portal).

## Before running the sample for the first time

1. Open an instance of PowerShell, Windows Terminal, Command Prompt or equivalent and navigate to the directory where you want to clone the sample.
2. git clone `https://github.com/Azure-Samples/communication-services-javascript-quickstarts.git`.
3. cd into the `chat-openai-sample` folder.
4. From the root of the `chat-openai-sample` folder run `npm install`.

### Setup and host your Azure DevTunnel

[Azure DevTunnels](https://learn.microsoft.com/en-us/azure/developer/dev-tunnels/get-started?tabs=windows) is an Azure service that enables you to share local web services hosted on the internet. Use the following commands to connect your local development environment to the public internet. This creates a tunnel with a persistent endpoint URL and enables anonymous access. We use this endpoint to notify your application of chat events from the Azure Communication Services Chat service.

> ```bash
> devtunnel create --allow-anonymous

Reviewer 3: Would it helpful to havesomething like this?

```suggestion
```bash
devtunnel login // Only needs to be done the first time
devtunnel create --allow-anonymous
```

devtunnel port create -p 8080
devtunnel host
```

### Configuring application

Open the `.env` file to configure the following settings

1. `PORT`: Localhost port to run the server app on.

> 1. `CONNECTION_STRING`: Azure Communication Service resource's connection string.

Reviewer 4: ```suggestion
1. `CONNECTION_STRING`: Azure Communication Services resource connection string.
```

> 1. `ACS_URL_ENDPOINT`: Azire Communication Service resource's url endpoint.

Reviewer 4: ```suggestion
1. `ACS_URL_ENDPOINT`: Azure Communication Services resource URL endpoint.
```

> 1. `AZURE_OPENAI_SERVICE_KEY`: Azure Open AI service key

Reviewer 4: ```suggestion
1. `AZURE_OPENAI_SERVICE_KEY`: Azure Open AI service key.
```

> 1. `AZURE_OPENAI_SERVICE_ENDPOINT`: Azure Open AI endpoint

Reviewer 4: ```suggestion
1. `AZURE_OPENAI_SERVICE_ENDPOINT`: Azure Open AI endpoint.
```

6. `AZURE_OPENAI_DEPLOYMENT_MODEL_NAME`: Azure Open AI deployment name.

> 

Reviewer 3: Maybe indicate where to get these

Author: Would that be too much hand holding? It should be part of prerequisite for "An Azure OpenAI Resource and Deployed Model. See [instructions](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/create-resource?pivots=web-portal)."

> 1. `AZURE_OPENAI_DEPLOYMENT_MODEL_NAME`: Azure Open AI deployment name

Reviewer 4: ```suggestion
1. `AZURE_OPENAI_DEPLOYMENT_MODEL_NAME`: Azure Open AI deployment name.
```

### Run app locally

> 1. Open a new Powershell window, cd into the `chat-openai-sample` folder and run `npm run dev`

Reviewer 4: ```suggestion
1. Open a new Powershell window, cd into the `chat-openai-sample` folder and run `npm run dev`.
```

> 1. Browser should pop up with the below page. If not navigate it to `http://localhost:8080/`

Reviewer 4: ```suggestion
1. The browser displays the following dialog box. If not navigate to `http://localhost:8080/`.
```

1. To test the AI analysis API endpoint on your local machine, in another new Powersehll window for the same directory, run `npm run client` to observe how messages are generated and processed.

> 1. Register an EventGrid Webhook for the `ChatMessageReceived` event that points to your DevTunnel URI for `<DevTunnelUri>/api/chatMessageReceived`.

Reviewer 2: For the Event Grid processing, it appears here as a step without any prior mention about it, so it might be confusing.

A couple of ideas:

1. Explicitly indicate it is part of the quickstart by describing it in the introduction and add prerequisite steps to set up Event Grid.
2. Have a separate optional section for all Event Grid related setup and what to look out for when the quickstart app is running.

Reviewer 3: Second this.

> 1. Register an EventGrid Webhook for the `ChatMessageReceived` event that points to your DevTunnel URI for `<DevTunnelUri>/api/chatMessageReceived`.

Author: do you mean the `1. `? the numbering doesn't really matter, when viewing the file will be automatically ordered to 1,2,3..., but if you want i can change it to ascending numbers

> 1. Register an EventGrid Webhook for the `ChatMessageReceived` Event that points to your DevTunnel URI for `<DevTunnelUri>/api/chatMessageReceived`.

Reviewer 4: ```suggestion
1. Register an EventGrid Webhook for the `ChatMessageReceived` event that points to your DevTunnel URI for `<DevTunnelUri>/api/chatMessageReceived`.
```
